audit/socket_auditor_test.go

```diff
@@ -0,0 +1,420 @@
+			}
+			expected := "https://servernotup" + strconv.Itoa(i) + ".com"
+			if resource.HttpRequest.Url != expected {
+				t.Errorf("expected log %d URL %s got %v", i, expected, resource.HttpRequest.Url)
+			}
+		}
+	case <-time.After(5 * time.Second):
+		t.Fatal("timeout waiting for retry flush")
+	}
+
+	// Trigger another batch and verify contents
+	for i := 0; i < auditor.batchSize-1; i++ {
+		url := "https://secondbatch" + strconv.Itoa(i) + ".com"
+		auditor.AuditRequest(Request{Method: "GET", URL: url, Allowed: true})
+	}
+
+	select {
+	case req := <-received:
+		if len(req.Logs) != auditor.batchSize {
+			t.Errorf("expected %d logs from retry, got %d", auditor.batchSize, len(req.Logs))
+		}
+		for i, log := range req.Logs {
+			resource, ok := log.Resource.(*agentproto.BoundaryLog_HttpRequest_)
+			if !ok {
+				t.Fatal("unexpected resource type")
+			}
+			expected := "https://secondbatch" + strconv.Itoa(i-1) + ".com"
+			if i == 0 {
+				expected = "https://serverup.com"
+			}
+			if resource.HttpRequest.Url != expected {
+				t.Errorf("expected log %d URL %s got %v", i, expected, resource.HttpRequest.Url)
+			}
+		}
+	case <-time.After(5 * time.Second):
+		t.Fatal("timeout waiting for flush")
+	}
+}
+
+func TestFlush_EmptyBatch(t *testing.T) {
+	t.Parallel()
+
+	err := flush(nil, nil)
+	if err != nil {
+		t.Errorf("expected nil error for empty batch, got %v", err)
+	}
+
+	err = flush(nil, []*agentproto.BoundaryLog{})
+	if err != nil {
+		t.Errorf("expected nil error for empty slice, got %v", err)
+	}
+}
+
+// tempDirUnixSocket returns a temporary directory that can safely hold unix
+// sockets (probably).
+//
+// During tests on darwin we hit the max path length limit for unix sockets
+// pretty easily in the default location, so this function uses /tmp instead to
+// get shorter paths.
+func tempDirUnixSocket(t *testing.T) string {
+	t.Helper()
+	if runtime.GOOS == "darwin" {
+		testName := strings.ReplaceAll(t.Name(), "/", "_")
+		dir, err := os.MkdirTemp("/tmp", testName)
+		if err != nil {
+			t.Errorf("failed to create temp dir: %v", err)
+		}
+
+		t.Cleanup(func() {
+			err := os.RemoveAll(dir)
+			if err != nil {
+				t.Fatalf("remove temp dir %s: %v", dir, err)
+			}
+		})
+		return dir
+	}
+
+	return t.TempDir()
+}
+
+func setupSocketAuditor(t *testing.T) *SocketAuditor {
+	socketPath := path.Join(tempDirUnixSocket(t), "server.sock")
+	logger := slog.New(slog.NewTextHandler(io.Discard, nil))
+	return &SocketAuditor{
+		socketPath:         socketPath,
+		logger:             logger,
+		logCh:              make(chan *agentproto.BoundaryLog, 2*defaultBatchSize),
+		batchSize:          defaultBatchSize,
+		batchTimerDuration: defaultBatchTimerDuration,
+	}
+}
+
+// startTestServer starts a Unix socket server that reads length-prefixed protobuf messages,
+// and reports all received requests to the given channel.
+func startTestServer(t *testing.T, socketPath string, received chan<- *agentproto.ReportBoundaryLogsRequest) {
```

Comment:
This is duplicating business logic from the agent.
Keeping these aligned will be tricky.
Either share the code or don't bleed upstream responsibility into this project.

Reply:
boundary now imports the `codec` package that has the shared framing implementation. I also adopted the use of `net.Pipe` in these tests to simplify.